types/next-auth/client.d.ts

```diff
@@ -46,21 +59,18 @@ declare function session(
 declare const getSession: typeof session;
 declare function csrfToken(context?: NextContext): Promise<string | null>;
 declare const getCsrfToken: typeof csrfToken;
+declare function signin(
+    provider: 'credentials' | 'email' | 'string',
+    data?: SignInDataParams,
+    authorizationParams?: SignInAuthorizationParams
+  ): Promise<SignInObjectReturn>;
 declare function signin(
     provider?: string,
-    data?: GenericObject & {
-        callbackUrl?: string;
-        redirect?: boolean;
-    },
-    authorizationParams?: string | string[][] | GenericObject | URLSearchParams
+    data?: SignInDataParams,
+    authorizationParams?: SignInAuthorizationParams
 ): Promise<void>;
 declare const signIn: typeof signin;
-declare function signout(
-    data?: {
-        callbackUrl?: string;
-        redirect?: boolean;
-    }
-): Promise<void>;
+declare function signout(data?: { callbackUrl?: string, redirect?: boolean }): Promise<void>;
```

Comment:
There is also `SignOutResponse` type in `next-auth`. Maybe not in scope of this PR, but would be nice to have:
https://github.com/nextauthjs/next-auth/blob/main/src/client/index.d.ts#L97

Reply:
In this case, these types are not a response. Maybe `SignOutDataParams` ?

Reply:
@bakazsorban44 could you clarify why the name is `SignOutResponse`?

Reply:
@balazsorban44 could you clarify why the name is `SignOutResponse`?

Reply:
probably by mistake. I would call it `SignOutResult` though.